staging/src/github.com/openshift/sdn/cmd/sdn-cni-plugin/openshift-sdn_linux.go

```diff
@@ -222,26 +222,25 @@ func (p *cniPlugin) CmdAdd(args *skel.CmdArgs) error {
 			if err != nil {
 				return fmt.Errorf("failed to configure macvlan device: %v", err)
 			}
+		}
+		var dsts []*net.IPNet
+		for _, addr := range addrs {
+			dsts = append(dsts, &net.IPNet{IP: addr.IP, Mask: net.CIDRMask(32, 32)})
+		}
 
-			var dsts []*net.IPNet
-			for _, addr := range addrs {
-				dsts = append(dsts, &net.IPNet{IP: addr.IP, Mask: net.CIDRMask(32, 32)})
-			}
+		_, serviceIPNet, err := net.ParseCIDR(config.ServiceNetworkCIDR)
+		if err != nil {
+			return fmt.Errorf("failed to parse ServiceNetworkCIDR: %v", err)
```

Comment:
Since we always parse `config.ServiceNetworkCIDR` now, how about moving it up to the top, right after `config` is read, so that if there is a problem, we don't bother half-setting-up the pod before we notice it.